tools/java/org/unicode/cldr/tool/ChartDelta.java

```diff
@@ -1008,14 +966,10 @@ private void addRow(Multimap<PathHeader, String> target, PathHeader key, String
 
         for (Pair<String, String> s : contents1) {
             String path = s.getFirst();
-            if (path.contains("/collat")) {
-                int debug = 0;
-            }
-
             String value = s.getSecond();
             if (dtdType == null) {
                 dtdType = DtdType.fromPath(path);
-                dtdData = DtdData.getInstance(dtdType);
+                dtdData = DtdData.getInstance(dtdType, CLDRConfig.getInstance().getCldrBaseDirectory());
```

Comment:
There are a lot of changes in this file that are mysterious to me, like this one. Please explain why you are doing these changes.

Reply:
> There are a lot of changes in this file that are mysterious to me, like this one. Please explain why you are doing these changes.

That's to avoid a deprecation warning.

Reply:
it avoids a static that assumes global state: But better would be for ChartDelta  to calculate its own `CLDRConfig.getInstance()` singleton in one place and use that everywhere.

Reply:
Agreed. Or in this case, getCldrBaseDirectory will not change during a session, so one could cache the directory in a static string

Reply:
OK, I'll cache it in a static string.